project/CLI.py

```diff
@@ -92,15 +105,41 @@ def __create(self, args):
             return "Game Created"
         return "Game Failed"
 
+    def __performTeamEdit(self, args):
+        if args[1] == "name":
+            if self.game.modify_team(args[2], args[4]):
+                return "Team name changed"
+        elif args[1] == "password":
+            if self.game.modify_team(args[2], args[4]):
+                return "Team password changed"
+        else:
+            return "Invalid command, check 2nd argument"
+
+    #bit complicated because both GM and teams can edit (their own for team), (GM -> any team)
+    def __editteam(self, args):
+        try:
+            if self.current_user.is_admin():
+                return self.__performTeamEdit(args)
+            else:
+                if self.current_user: # if current user is a some team and not game maker
+                    if self.current_user.get_username() == args[2]: # loggedin team can only edit themselves
+                        return self.__performTeamEdit(args)
+        except IndexError:
+            return "Invalid Parameters"
+        except AttributeError:
+            return "Have to be logged in to Edit team"
+
     def command(self, args):
         commands = {"login": self.__login, "addteam": self.__addteam, "addlandmark": self.__addlandmark,
                     "removeteam": self.__removeteam, "start": self.__start, "create": self.__create,
-                    "logout": self.__logout}
+                    "logout": self.__logout, "editteam": self.__editteam, "removelandmark": self.__removelandmark}
         inp = shlex.split(args)
         try:
             return commands[inp[0].lower()](inp)
-        except (KeyError, IndexError):
+        except KeyError:
             return "Invalid command"
+        except IndexError:
```

Comment:
If we hit the IndexError here then there was an empty command passed, which is kinda an invalid command, not invalid parameters... Not a big deal though.